static/app/components/modals/insightChartModal.tsx

```diff
@@ -13,7 +13,20 @@ export type InsightChartModalOptions = {
 };
 type Props = ModalRenderProps & InsightChartModalOptions;
 
-export default function InsightChartModal({Header, title, children}: Props) {
+export default function InsightChartModal({
+  Header,
+  title,
+  children,
+  modalContainerRef,
+}: Props) {
+  useEffect(() => {
+    // display the fullscreen insight charts on top of any sidebar elements
+    // with z-index 10000 or 10001
```

Comment:
I'm kind of suprised we have to do this, typically a modal should always be on top, no matter what what. If we have to manually update the zIndex for the modal, i feel as though the sidebar's zIndex is too high?

Reply:
the modal has z-index 10000 but the sidebar has z-index 10001. I think this is the only place in the product where we possibly display both a modal and a sidebar

Reply:
Could we make the sidebar zIndex 9999?

Reply:
We would be future proofing ourselves here, there may be more cases for sidebar and modal at some point